cpp/src/arrow/array.h

```diff
@@ -709,6 +709,14 @@ class ARROW_EXPORT FixedSizeListArray : public Array {
     return values_->Slice(value_offset(i), value_length(i));
   }
 
+  /// \brief Construct FixedSizeListArray from child value array and value_length
+  ///
+  /// \param[in] values Array containing list values
+  /// \param[in] list_size The fixed length of each list
+  /// \param[out] out Will have length equal to values.length() / list_size
+  static Status FromArrays(const std::shared_ptr<Array>& values, int32_t list_size,
```

Comment:
Since we're discussing migrating public APIs to `Result<>`, this one should probably return a `Result<std::shared_ptr<Array>>`.